v19.1/architecture/replication-layer.md

```diff
@@ -69,13 +69,15 @@ To achieve this, each lease renewal or transfer also attempts to collocate them.
 
 #### Epoch-based leases (table data)
 
-To manage leases for table data, CockroachDB implements a notion of "epochs," which are defined as the period between a node joining a cluster and a node disconnecting from a cluster. When the node disconnects, the epoch is considered changed, and the node immediately loses all of its leases.
+To manage leases for table data, CockroachDB implements a notion of "epochs," which are defined as the period between a node joining a cluster and a node disconnecting from a cluster. To extend its leases, each node must periodically update its liveness record, which is stored on a system range key. When a node disconnects, it stops updating the liveness record, and the the epoch is considered changed. This causes the node to immediately lose all of its leases.
 
 This mechanism lets us avoid tracking leases for every range, which eliminates a substantial amount of traffic we would otherwise incur. Instead, we assume leases do not expire until a node loses connection.
```

Comment:
The language around "lets us avoid tracking leases for every range" is actually what originally tripped me up. We still _do_ track leases for every range, but we're smarter about it. By assuming leases don't expire until a node loses connection, each leaseholder does not have to individually renew their own lease. By tying lease lifetimes to the node itself (that holds many many leaseholders), we eliminate a substantial amount of traffic and Raft processing we would otherwise incur. I'd include something to this effect here instead. Everything else LGTM.